python-sdk/src/astro/files/locations/base.py

```diff
@@ -188,3 +189,6 @@ def get_stream(self):
         :param df: pandas dataframe
         """
         return smart_open.open(self.smartopen_uri, mode="wb", transport_params=self.transport_params)
+
+    def get_snowflake_stage_auth_sub_statement(self) -> str:
```

Comment:
Is this error relevant for every file location?

Reply:
I think this is safe a default at the moment. Since this methods is only called for native transfer 
and  2/3 (Azure and GCS) needs to raise this error if the `storage_integration` is not passed.